src/index.js

```diff
@@ -37,248 +34,105 @@ export default function supernova(env) {
       data,
     },
     component: () => {
-      const [objectData, setObjectData] = useState(null);
-      const [styling, setStyling] = useState(null);
-      const [expandedState, setExpandedState] = useState(null);
-      const [linked, setLinked] = useState(false);
-      const [selectionState, setSelectionState] = useState([]);
-      const [transform, setTransform] = useState(null);
-      const [initialZoom, setInitialZoom] = useState(null);
       const layout = useStaleLayout();
       const model = useModel();
       const element = useElement();
       const Theme = useTheme();
       const options = useOptions();
-      const [opts] = useState(options);
       const rect = useRect();
-      const selectionsAPI = useSelections();
       const constraints = useConstraints();
-      const [selectionsAndTransform] = useState({
-        api: selectionsAPI,
-        setState: setSelectionState,
-        linked: false,
+      const translator = useTranslator();
+      const [preRenderData, setPreRenderData] = useState(null);
+      const [styling, setStyling] = useState(null);
+      const [expandedState, setExpandedState] = useState(null);
+      const [transform, setTransform] = useState(null);
+      const [initialZoom, setInitialZoom] = useState(null);
+      const [storageState] = useState({
         transform: {},
         constraints,
       });
+      const selectionObj = selectionHandler(translator);
 
-      const translator = useTranslator();
       useEffect(() => {
         autoRegister(translator);
-      }, [translator]);
-
-      const resetSelections = () => {
-        setSelectionState([]);
-      };
-      const resetSelectionsAndLinked = () => {
-        setLinked(false);
-        setSelectionState([]);
-      };
+      }, [translator.language()]);
```

Comment:
Did you find that updating the language doesn't work?

Reply:
Yes, it works similar to how constraints work. Miralem told me this a while back, came to mind now.

Reply:
I think the language only gets updated in Sense after a full refresh right?